src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogSenderHelper.java

```diff
@@ -148,18 +164,37 @@ public static void sendPipelineStages(List<PipelineStageModel> stages, BuildMode
         data.put("number", buildModel.getNumber());
 
         if (CollectionUtils.isNotEmpty(stages)) {
-            //20 Stages at a time
-            List<List<PipelineStageModel>> partition = Lists.partition(stages, NUMBER_OF_STAGES);
-            for(List<PipelineStageModel> stageModels : partition){
-                data.put("stages", stageModels);
-                LogSender.getInstance().sendLogs(pluginDescriptor.getUrl(), gson.toJson(data).getBytes()
-                        , null, pluginDescriptor.getSourceCategory());
+            // send Stages based on the size
+            List<PipelineStageModel> toBeSent = new LinkedList<>();
+            int size = gson.toJson(data).getBytes().length;
+            for (PipelineStageModel pipelineStageModel : stages) {
+                size = size + gson.toJson(pipelineStageModel).getBytes().length;
+                if (size > MAX_DATA_SIZE) {
+                    sendStagesInChunksOfPreDefinedSize(buildModel, gson, pluginDescriptor, toBeSent, data);
+                    toBeSent.clear();
+                }
+                toBeSent.add(pipelineStageModel);
+                data.put("stages", toBeSent);
```

Comment:
This can be done out of the for loop as we are not recreating the toBeSent list.

Reply:
Yes. But as soon the size reaches 50 KB we sent that data and then clear the toBeSent list. Then the later elements get added into the list again and the loop continues.

Reply:
I have moved the put code into the method when we actually send the data. and we are now checking the size of the toBeSent list.

Reply:
Done